File: SqrtNewtonRaphson/C/SqrtNewtonRaphson.c
#include <stdio.h>

double sqrtNR(double number, double precision)
{
	double a = number; 
	while (a - number / a > precision)
		a = (a + number / a) / 2;
	return a;
}

int main()
{
	printf("%.50f\n", sqrtNR(737, 0.0001));

	return 0;
}

File: SqrtNewtonRaphson/Go/SqrtNewtonRaphson.go
package main

import "fmt"

func sqrtNR(number float64, precision float64) float64 {
	
	var result = number
	for result - number / result > precision {
		
		result = (result + number / result) / 2

	}

	return result
}

func main() {
	
	fmt.Println(sqrtNR(737, 0.0001))
}
